src/bam/collapse_reads_to_fragments/pipeline.rs

```diff
@@ -271,33 +284,46 @@ pub fn calc_consensus_complete_groups<'a, W: io::Write>(
 
         if !r_recs.is_empty() {
             let overlap = calc_overlap(&l_recs[0], &r_recs[0])?;
-            if overlap > 0 {
-                let uuid = &Uuid::new_v4().to_hyphenated().to_string();
-                l_seqids.append(&mut r_seqids);
-                fq_se_writer.write_record(
-                    &CalcOverlappingConsensus::new(
-                        &l_recs,
-                        &r_recs,
-                        overlap as usize,
-                        &l_seqids,
-                        uuid,
-                        verbose_read_names,
-                    )
-                    .calc_consensus()
-                    .0,
-                )?;
-            } else {
-                let uuid = &Uuid::new_v4().to_hyphenated().to_string();
-                fq1_writer.write_record(
-                    &CalcNonOverlappingConsensus::new(&l_recs, &l_seqids, uuid)
+            match (
+                overlap > 0,
+                is_valid_overlap(
+                    overlap as u32,
+                    l_recs[0].cigar_cached().unwrap().into_iter().rev(),
```

Comment:
So, this is only checked for the first record in l_recs and r_recs? What about the rest?

Reply:
We previously only calculated the overlap just for the first record of r1 and r2 as we expected reads of the same duplicate group being structurally equal.